Gems/Compression/Code/Include/Compression/CompressionInterfaceAPI.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/std/containers/span.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <Compression/CompressionInterfaceStructs.h>
+
+namespace Compression
+{
+    enum class CompressionResult : AZ::u8
+    {
+        Started,
+        Complete,
+        NotStarted,
+        Failed,
+    };
+
+    struct CompressionResultData
+    {
+        //! Returns a boolean true if decompression has completed
+        explicit constexpr operator bool() const
+        {
+            return m_result == CompressionResult::Complete;
+        }
+
+        //! Retrieves the compressed data size
+        AZ::u64 GetBytesCompressed() const
+        {
+            return m_compressedBuffer.size();
+        }
+
+        //! Retrieves the memory address of the compressed data
+        AZStd::byte* GetAddress() const
```

Comment:
Nitpick: To be consistent with the previous function it should including some reference to "compressed" such as `GetCompressedData`.

Reply:
How about `GetCompressedByteData`?

That way it will be consistent with `GetCompressedByteCount`